tests/rptest/tests/schema_registry_test.py

```diff
@@ -1734,6 +1768,152 @@ def check_each_schema(subject: str, schemas: list[str],
                               subjects[subject]["subject_versions"])
 
 
+class SchemaRegistryModeNotMutableTest(SchemaRegistryEndpoints):
+    """
+    Test that mode cannot be mutated when mode_mutability=False.
+    """
+    def __init__(self, context, **kwargs):
+        self.schema_registry_config = SchemaRegistryConfig()
+        self.schema_registry_config.mode_mutability = False
+
+        super(SchemaRegistryEndpoints, self).__init__(
+            context,
+            schema_registry_config=self.schema_registry_config,
+            **kwargs)
+
+    @cluster(num_nodes=3)
+    def test_mode_immutable(self):
+
+        subject = f"{create_topic_names(1)[0]}-key"
+
+        result_raw = self._get_mode()
+        assert result_raw.status_code == 200
+        assert result_raw.json()["mode"] == "READWRITE"
+
+        result_raw = self._set_mode(data=json.dumps({"mode": "INVALID"}))
+        assert result_raw.status_code == 422
+        assert result_raw.json()["error_code"] == 42204
+
+        result_raw = self._set_mode(data=json.dumps({"mode": "READONLY"}))
+        assert result_raw.status_code == 422
+        assert result_raw.json()["error_code"] == 42205
+        assert result_raw.json()["message"] == "Mode changes are not allowed"
+
+        result_raw = self._set_mode_subject(subject=subject,
+                                            data=json.dumps(
+                                                {"mode": "READONLY"}))
+        assert result_raw.status_code == 422
+        assert result_raw.json()["error_code"] == 42205
+        assert result_raw.json()["message"] == "Mode changes are not allowed"
+
+        result_raw = self._delete_mode_subject(subject=subject)
+
+
+class SchemaRegistryModeMutableTest(SchemaRegistryEndpoints):
+    """
+    Test schema registry mode against a redpanda cluster.
+    """
+    def __init__(self, context, **kwargs):
+        self.schema_registry_config = SchemaRegistryConfig()
+        self.schema_registry_config.mode_mutability = True
+        super(SchemaRegistryEndpoints, self).__init__(
+            context,
+            schema_registry_config=self.schema_registry_config,
+            **kwargs)
+
+    @cluster(num_nodes=3)
+    def test_mode(self):
+        """
+        Smoketest mode endpoints
+        """
+        subject = f"{create_topic_names(1)[0]}-key"
+        not_subject = f"{create_topic_names(1)[0]}-key"
+
+        self.logger.debug("Get initial global mode")
+        result_raw = self._get_mode()
+        assert result_raw.status_code == 200
+        assert result_raw.json()["mode"] == "READWRITE"
+
+        self.logger.debug("Set invalid global mode")
+        result_raw = self._set_mode(data=json.dumps({"mode": "INVALID"}))
+        assert result_raw.status_code == 422
+        assert result_raw.json()["error_code"] == 42204
```

Comment:
Is the message the same as 1907-1916? Or is there something to distinguish "nonsense mode" vs "valid mode that is unsupported"?

Reply:
> Is the message the same as 1907-1916? Or is there something to distinguish "nonsense mode" vs "valid mode that is unsupported"?

You're sharp (it was clearly intended and not just copy-pasta, honest)

Reply:
There is no error code for unsupported upstream, so I just chose something.

Reply:
yeah that's what I figured. just curious